core/src/main/scala/kafka/coordinator/transaction/TransactionMetadata.scala

```diff
@@ -109,7 +125,7 @@ private[transaction] class TransactionMetadata(val transactionalId: String,
   def prepareIncrementProducerEpoch(newTxnTimeoutMs: Int,
                                     expectedProducerEpoch: Option[Short],
                                     updateTimestamp: Long): Either[Errors, TxnTransitMetadata] = {
-    if (isProducerEpochExhausted)
+    if (TransactionMetadata.isEpochExhausted(producerEpoch))
```

Comment:
This changed because `isProducerEpochExhausted` used `producerEpoch`, but now it uses `clientProducerEpoch` (which sometimes could be `nextProducerEpoch`), so in the place we want to continue checking `producerEpoch`, thus this is an explicit call.

Reply:
To clarify for myself:
This method would not be called from keepPrepared=True initProducerId calls, and thus we should update the ongoing transaction's producer epoch and not the client (in some cases nextProducerEpoch) epoch.

If we call this method, how does that affect nextProducerEpoch/Id? I could imagine we bump the epoch here such that it matches the nextProducerEpoch.